service/controller/resource/app/desired.go

```diff
@@ -55,11 +50,17 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*g8s
 		return nil, microerror.Mask(err)
 	}
 
+	componentVersions, err := r.releaseVersion.ComponentVersion(ctx, cr)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	appOperatorVersion := componentVersions[label.AppOperator]
```

Comment:
Seeing this I am wondering if we would be better off with an interface like this. That maybe makes sense for operators but maybe not for the apps depending on how this is used. Uncertain. Using the map like this means you need trust in the string map returned. And then having some label from some other package have to work with it together makes it more fragile to use. 

```
v, err := r.releaseVersion.AppOperator(ctx, cr)
```